File: 025. First Non-Repeating Character/FirstNonRepeatingCharacter.java
import java.util.*;

class FirstNonRepeatingCharacter {

    public int firstNonRepeatingCharacter(String string) {
        HashMap<Character, Integer> characterFrequencies = new HashMap<Character, Integer>();
        for (int i = 0; i < string.length(); i++) {
            char character = string.charAt(i);
            characterFrequencies.put(character, characterFrequencies.getOrDefault(character, 0) + 1);
        }
        for (int i = 0; i < string.length(); i++) {
            char character = string.charAt(i);
            if (characterFrequencies.get(character) == 1) {
                return i;
            }
        }
        return -1;
    }
}

File: 025. First Non-Repeating Character/FirstNonRepeatingCharacter.kt
package com.algoexpert.program

fun firstNonRepeatingCharacter(string: String): Int {
    val characterFrequencies = mutableMapOf<Char, Int>()
    for (char in string) {
        if (!(char in characterFrequencies)) characterFrequencies[char] = 0
        characterFrequencies[char] = characterFrequencies[char]!! + 1
    }
    for (i in 0 until string.length) {
        val char = string[i]
        if (characterFrequencies[char] == 1) {
            return i
        }
    }
    return -1
}
